### PR TITLE
Add shared-memory access to local database

### LINKED ISSUE #52: Add support for shared-memory access to chaindata
Add support for shared-memory access to chaindata as replacement for `KV` interface in local deployments (`Silkrpc` colocated with `TurboGeth Core`)

### PR DESCRIPTION
**Pull request by lupin012**
Closes #52

### CODE REVIEW

**Review by canepat — CHANGES_REQUESTED** [review]

File: cmd/silkrpc_daemon.cpp [hunk 1]
```diff
@@ -42,6 +42,7 @@ ABSL_FLAG(uint32_t, timeout, silkrpc::kDefaultTimeout.count(), "gRPC call timeou
 ABSL_FLAG(silkrpc::LogLevel, log_verbosity, silkrpc::LogLevel::Critical, "logging verbosity level");
 ABSL_FLAG(silkrpc::WaitMode, wait_mode, silkrpc::WaitMode::blocking, "scheduler wait mode");
 ABSL_FLAG(std::string, jwt_secret_file, silkrpc::kDefaultJwtFilename, "Token file to ensure safe connection between CL and EL");
+ABSL_FLAG(std::string, db_path, silkrpc::kDefaultDbPath, "DB Path");
```
[the remark this answers is not in the document]
Comment (lupin012) [reply cmd/silkrpc_daemon.cpp:45]: ok

File: silkworm/silkrpc/common/constants.hpp [hunk 2]
```diff
@@ -42,6 +42,7 @@ constexpr const char* kDefaultEnginePort{"localhost:8551"};
 constexpr const char* kDefaultTarget{"localhost:9090"};
 constexpr const char* kDefaultEth1ApiSpec{"debug,eth,net,parity,erigon,trace,web3,txpool"};
 constexpr const char* kDefaultEth2ApiSpec{"engine,eth"};
+constexpr const char* kDefaultDbPath{""};
```
[the remark this answers is not in the document]
Comment (lupin012) [reply silkworm/silkrpc/common/constants.hpp:45]: ok

File: silkworm/silkrpc/daemon.hpp [hunk 3]
```diff
@@ -42,6 +42,7 @@ struct DaemonSettings {
     LogLevel log_verbosity;
     WaitMode wait_mode;
     std::string jwt_secret_filename;
+    std::string db_path;
```
[the remark this answers is not in the document]
Comment (lupin012) [reply silkworm/silkrpc/daemon.hpp:45]: ok

File: silkworm/silkrpc/concurrency/context_pool.cpp [hunk 4]
```diff
@@ -35,16 +36,23 @@ Context::Context(
...
       state_cache_(state_cache),
+      chaindata_env_(chaindata_env),
       wait_mode_(wait_mode) {
     std::shared_ptr<grpc::Channel> channel = create_channel();
-    database_ = std::make_unique<ethdb::kv::RemoteDatabase>(*grpc_context_, channel);
+    if (!db_path.empty()) {
+        database_ = std::make_unique<ethdb::file::LocalDatabase>(chaindata_env);
+    } else {
+        database_ = std::make_unique<ethdb::kv::RemoteDatabase>(*grpc_context_, channel);
+    }
```
[the remark this answers is not in the document]
Comment (lupin012) [reply silkworm/silkrpc/concurrency/context_pool.cpp:55]: ok

File: silkworm/silkrpc/ethdb/cursor.hpp [hunk 5]
```diff
@@ -89,5 +89,30 @@ class SplitCursor {
     SplittedKeyValue split_key_value(const KeyValue& kv);
 };
 
+class SplitDupSortCursor {
```
[the remark this answers is not in the document]
Comment (lupin012) [reply silkworm/silkrpc/ethdb/cursor.hpp:92]: ok renamed

File: silkworm/silkrpc/ethdb/split_dup_sort_cursor.cpp [hunk 6]
```diff
@@ -0,0 +1,91 @@
+/*
```
[the remark this answers is not in the document]
Comment (lupin012) [reply silkworm/silkrpc/ethdb/split_dup_sort_cursor.cpp:1]: ok moved to cursor.cpp

**Review by canepat — APPROVED** [review]

**Pull request merged**